File: scala.leap.motion/src/tutorial/scalaone/SampleListener.scala
package tutorial.scalaone

import com.leapmotion.leap.{ Controller, Frame, Listener }

import utils.Utils

class SampleListener extends Listener {
  override def onInit(controller: Controller) {
    if ("true" == System.getProperty("lm.debug", "false")) println("Invoking " + Utils.caller(new Throwable))
    super.onInit(controller)
  }

  override def onConnect(controller: Controller) {
    if ("true" == System.getProperty("lm.debug", "false")) println("Invoking " + Utils.caller(new Throwable))
    println("Connected")
  }

  override def onDisconnect(controller: Controller) {
    if ("true" == System.getProperty("lm.debug", "false")) println("Invoking " + Utils.caller(new Throwable))
    super.onDisconnect(controller)
  }

  override def onExit(controller: Controller) {
    if ("true" == System.getProperty("lm.debug", "false")) println("Invoking " + Utils.caller(new Throwable))
    super.onExit(controller)
  }

  override def onFrame(controller: Controller) {
    if ("true" == System.getProperty("lm.debug", "false")) println("Invoking " + Utils.caller(new Throwable))
    val frame: Frame = controller.frame
    println(frame.toString)
  }

  override def onFocusGained(controller: Controller) {
    if ("true" == System.getProperty("lm.debug", "false")) println("Invoking " + Utils.caller(new Throwable))
    super.onFocusGained(controller)
  }

  override def onFocusLost(controller: Controller) {
    if ("true" == System.getProperty("lm.debug", "false")) println("Invoking " + Utils.caller(new Throwable))
    super.onFocusLost(controller)
  }

  override def onServiceConnect(controller: Controller) {
    if ("true" == System.getProperty("lm.debug", "false")) println("Invoking " + Utils.caller(new Throwable))
    super.onServiceConnect(controller)
  }

  override def onServiceDisconnect(controller: Controller) {
    if ("true" == System.getProperty("lm.debug", "false")) println("Invoking " + Utils.caller(new Throwable))
    super.onServiceDisconnect(controller)
  }

  override def onDeviceChange(controller: Controller) {
    if ("true" == System.getProperty("lm.debug", "false")) println("Invoking " + Utils.caller(new Throwable))
    super.onDeviceChange(controller)
  }
}

File: java.leap.motion/src/tutorial/one/SampleListener.java
package tutorial.one;

import com.leapmotion.leap.*;
import utils.Utils;

public class SampleListener extends Listener {

  @Override
  public void onInit(Controller controller)
  {
    if ("true".equals(System.getProperty("lm.debug", "false")))
      System.out.println("Invoking " + Utils.caller(new Throwable()));
    super.onInit(controller);
  }

  @Override
  public void onConnect(Controller controller) {
    if ("true".equals(System.getProperty("lm.debug", "false")))
      System.out.println("Invoking " + Utils.caller(new Throwable()));
    System.out.println("Connected");
  }

  @Override
  public void onDisconnect(Controller controller)
  {
    if ("true".equals(System.getProperty("lm.debug", "false")))
      System.out.println("Invoking " + Utils.caller(new Throwable()));
    super.onDisconnect(controller);
  }

  @Override
  public void onExit(Controller controller)
  {
    if ("true".equals(System.getProperty("lm.debug", "false")))
      System.out.println("Invoking " + Utils.caller(new Throwable()));
    super.onExit(controller);
  }

  @Override
  public void onFrame(Controller controller) {
    if ("true".equals(System.getProperty("lm.debug", "false")))
      System.out.println("Invoking " + Utils.caller(new Throwable()));
//  System.out.println("Frame available");
    Frame frame = controller.frame();
    System.out.println(frame.toString());
  }

  @Override
  public void onFocusGained(Controller controller)
  {
    if ("true".equals(System.getProperty("lm.debug", "false")))
      System.out.println("Invoking " + Utils.caller(new Throwable()));
    super.onFocusGained(controller);
  }

  @Override
  public void onFocusLost(Controller controller)
  {
    if ("true".equals(System.getProperty("lm.debug", "false")))
      System.out.println("Invoking " + Utils.caller(new Throwable()));
    super.onFocusLost(controller);
  }

  @Override
  public void onServiceConnect(Controller controller)
  {
    if ("true".equals(System.getProperty("lm.debug", "false")))
      System.out.println("Invoking " + Utils.caller(new Throwable()));
    super.onServiceConnect(controller);
  }

  @Override
  public void onServiceDisconnect(Controller controller)
  {
    if ("true".equals(System.getProperty("lm.debug", "false")))
      System.out.println("Invoking " + Utils.caller(new Throwable()));
    super.onServiceDisconnect(controller);
  }

  @Override
  public void onDeviceChange(Controller controller)
  {
    if ("true".equals(System.getProperty("lm.debug", "false")))
      System.out.println("Invoking " + Utils.caller(new Throwable()));
    super.onDeviceChange(controller);
  }
}